### PR TITLE
#260 - Checking for conflict in RpmUpload

### LINKED ISSUE #260: RpmUpload.java:48-53: Finish implementation of RpmUpload...
The puzzle `253-431eeb6b` from #253 has to be resolved: 

https://github.com/artipie/rpm-adapter/blob/13730e8292b7b1d0f64b64f58fae8670429777c4/src/main/java/com/artipie/rpm/http/RpmUpload.java#L48-L53

The puzzle was created by @olenagerasimova on 11-Jun-20. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #260 
Checking for conflict in RpmUpload and override param support

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#270](https://github.com/artipie/rpm-adapter/issues/270) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/5fa095be-833a-40f3-899b-7173e74509b8); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/rpm/http/RpmUpload.java [hunk 1]
```diff
@@ -81,15 +84,33 @@
...
+            conflict = this.asto.exists(key);
+        }
         return new AsyncResponse(
-            SingleInterop.fromFuture(
-                this.asto.save(
-                    request.file(),
-                    new Content.From(body)
-                ).thenApply(ignored -> true)
+            conflict.thenApply(
+                isConflict -> {
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/rpm/http/RpmUpload.java:96]: @olegmoz I'd suggest to rename `isConflict` to verb `conflicts`

File: src/main/java/com/artipie/rpm/http/RpmUpload.java [hunk 2]
```diff
@@ -81,15 +84,33 @@
     public Response response(final String line, final Iterable<Map.Entry<String, String>> headers,
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/rpm/http/RpmUpload.java:84]: @olegmoz please remove puzzle as this is last part of the job

**Comment by olegmoz** [comment]
@olenagerasimova thanks, made the changes

**Comment by 0crat** [comment]
Job was finished in 23 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Pull request merged**